elyra/pipeline/pipeline.py

```diff
@@ -169,6 +172,86 @@ def _scrub_list(dirty: Optional[List[Optional[str]]]) -> List[str]:
     def is_generic_operation(operation_type) -> bool:
         return True if operation_type in Operation.generic_node_types else False
 
+    @staticmethod
+    def process_dictionary_value(value: str, logger: Optional[Logger] = None) -> Optional[Dict]:
+        """
+        For parameters of type dictionary, convert the string value given in the pipeline
+        JSON to the appropriate Dict format. If Dict cannot be formed, log & return None.
+        """
+        if not value:
+            value = "{}"
+
+        converted_dict = None
+        if value.startswith('{') and value.endswith('}'):
+            try:
+                converted_dict = ast.literal_eval(value)
+            except Exception:
+                try:
+                    value = value.replace("'", "\"")
+                    converted_dict = json.loads(value)
+                except Exception:
+                    value = value[1:-1]
+
+        # Value could not be successfully converted to dictionary
+        if not isinstance(converted_dict, dict):
+            converted_dict = {}
+            kv_regex = re.compile(",(?=(?:[^]}]*[[{][^]}]*[]}])*[^]}]*$)")
+            kv_pairs = kv_regex.split(value)
+
+            for pair in kv_pairs:
+                key, val = pair.split(":", 1)
+
+                try:
+                    val = ast.literal_eval(val.strip(" '\""))
+                except Exception:
+                    try:
+                        val = json.loads(val.strip(" '\""))
+                    except Exception:
+                        pass
+
+                if isinstance(val, str):
+                    val = val.strip(" '\"")
+                    if val.startswith('{') and val.endswith('}'):
+                        val = Operation.process_dictionary_value(val, logger=logger)
+                    elif val.startswith('[') and val.endswith(']'):
+                        val = Operation.process_list_value(val)
+
+                converted_dict[key.strip(" '\"")] = val
+
+        if not isinstance(converted_dict, dict):
+            Operation._log_info(f"Could not convert parameter value '{value}' to dictionary",
```

Comment:
I need to determine if this case is ever reached with some of the latest changes I made. If not, will need to remove this block, change the type hints, and remove the logic in the processors that sends the user-entered string to the appropriate constructor if a dictionary could not be formed.

Reply:
I agree that it's impossible for `converted_dict` to not be a `dict` at this point by virtue of L197 and recommend the cleanup that is suggested.

I suppose you could, however, wind up with an empty dictionary when the input value is non-None and non-dict.  Perhaps that's the case in which a fallback to the input value should occur.  Hmm, in that case, you probably want to check the recursed call's return value against `None` so that you can break out of the `for` loop.

I also wonder if, in these situations, we just return the input value rather than `None` and change the return type to `Union[dict, str]` along with logging a **warning** (rather than info).  Of course, this would mean that the recursed return value would check for a string instance rather than `None` to determine if `break` is required...
```python
val = Operation.process_dictionary_value(val, logger=logger)
if isinstance(val, str):
    break
```

I think a test that takes a vanilla string value (that doesn't contain a colon) would be a good test.  Based on the above discussion, that test should return the input, and log a warning (although the latter can be difficult to verify).